### PR TITLE
Use instance vars in pipeline

### PR DESCRIPTION
**Pull request by hursey013**
## Changes proposed in this pull request:
- Support multiple envs within the single pipeline config file

## security considerations
None

### CODE REVIEW

File: ci/pipeline.yml [hunk 1]
```diff
@@ -355,27 +185,27 @@ jobs:
...
       - get: src
-        resource: src-staging
+        resource: src-((deploy-env))
         params: {depth: 1}
         trigger: true
       - set_pipeline: build-container
         file: src/ci/pipeline.yml
 
-  - name: reconfigure-production
+  - name: reconfigure-((deploy-env))
```
Comment (apburnes) [inline ci/pipeline.yml:204]: This is a duplicate job from the one above.

File: ci/pipeline.yml [hunk 2]
```diff
@@ -355,27 +185,27 @@ jobs:
...
             text: |
-              :white_check_mark: SUCCESS: Successfully deployed pages build containers on production
+              :white_check_mark: SUCCESS: Successfully deployed pages build containers on ((deploy-env))
               <$ATC_EXTERNAL_URL/teams/$BUILD_TEAM_NAME/pipelines/$BUILD_PIPELINE_NAME/jobs/$BUILD_JOB_NAME/builds/$BUILD_NAME|View build details>
             channel: ((slack-channel))
             username: ((slack-username))
             icon_url: ((slack-icon-url))
 
-  - name: reconfigure-staging
+  - name: reconfigure-((deploy-env))
```
Comment (apburnes) [inline ci/pipeline.yml:194]: I started calling the job `set-pipeline-((deploy-env))` but just realized we've used the `reconfigure` naming convention before. Whichever we want, let's stick to one.

**Review by apburnes — COMMENTED** [review]
Just pointed out some clean up comments/questions. Also, we never had a `pr` resource to run tests on PR. We'll want that job setup as well. Here's an example of [requiring the resource type](https://github.com/18F/federalist-proxy/blob/staging/ci/pipeline.yml#L303-L306), then [setting the resource](https://github.com/18F/federalist-proxy/blob/staging/ci/pipeline.yml#L244-L252), and then [running the tests](https://github.com/18F/federalist-proxy/blob/staging/ci/pipeline.yml#L116-L149).

File: ci/pipeline.yml (hunk 2 above)
Comment (hursey013) [reply ci/pipeline.yml:194]: `set-pipeline` seems much clearer - I'll update to that 👍

**Review by apburnes — APPROVED** [review]
🥳

**Pull request merged**